microprofile/openapi/src/main/java/io/helidon/microprofile/openapi/MpOpenApiFeature.java

```diff
@@ -0,0 +1,254 @@
+     *
+     * @param config {@code OpenApiConfig} object describing paths, servers, etc.
+     * @param staticFile the static file, if any, to be included in the resulting model
+     * @param filteredIndexViews possibly empty list of FilteredIndexViews to use in harvesting definitions from the code
+     * @return the OpenAPI model
+     * @throws RuntimeException in case of errors reading any existing static
+     * OpenAPI document
+     */
+    private OpenAPI prepareModel(OpenApiConfig config, OpenApiStaticFile staticFile,
+                                 List<? extends IndexView> filteredIndexViews) {
+        try {
+            // The write lock guarding the model has already been acquired.
+            OpenApiDocument.INSTANCE.reset();
+            OpenApiDocument.INSTANCE.config(config);
+            OpenApiDocument.INSTANCE.modelFromReader(OpenApiProcessor.modelFromReader(config, contextClassLoader()));
+            if (staticFile != null) {
+                OpenApiDocument.INSTANCE.modelFromStaticFile(OpenApiParser.parse(PARSER_HELPER.get().types(),
+                                                                                 staticFile.getContent()));
+            }
+            if (isAnnotationProcessingEnabled(config)) {
+                expandModelUsingAnnotations(config, filteredIndexViews);
+            } else {
+                LOGGER.log(System.Logger.Level.TRACE, "OpenAPI Annotation processing is disabled");
+            }
+            OpenApiDocument.INSTANCE.filter(OpenApiProcessor.getFilter(config, contextClassLoader()));
+            OpenApiDocument.INSTANCE.initialize();
+            OpenAPIImpl instance = OpenAPIImpl.class.cast(OpenApiDocument.INSTANCE.get());
+
+            // Create a copy, primarily to avoid problems during unit testing.
+            // The SmallRye MergeUtil omits the openapi value, so we need to set it explicitly.
+            return MergeUtil.merge(new OpenAPIImpl(), instance)
+                    .openapi(instance.getOpenapi());
+        } catch (IOException ex) {
+            throw new RuntimeException("Error initializing OpenAPI information", ex);
+        }
+    }
+
+
+    private static Format toFormat(OpenAPIMediaType openAPIMediaType) {
+        return openAPIMediaType.equals(OpenAPIMediaType.YAML)
+                ? Format.YAML
+                : Format.JSON;
+    }
+
+    private boolean isAnnotationProcessingEnabled(OpenApiConfig config) {
+        return !config.scanDisable();
+    }
+
+    private void expandModelUsingAnnotations(OpenApiConfig config, List<? extends IndexView> filteredIndexViews) {
+        if (filteredIndexViews.isEmpty() || config.scanDisable()) {
+            return;
+        }
+
+        /*
+         * Conduct a SmallRye OpenAPI annotation scan for each filtered index view, merging the resulting OpenAPI models into one.
+         * The AtomicReference is effectively final so we can update the actual reference from inside the lambda.
```

Comment:
True. Although this code is "new" because it's in a new class, I transplanted it from another class unchanged so it's "not new" in that sense, and I'm tempted to leave it alone. Luckily this is not performance-critical code but I'll keep this technique in mind for the future. Thanks.